designer/server/src/config.ts

```diff
@@ -1,130 +1,167 @@
+  AZURE_CLIENT_ID,
+  AZURE_CLIENT_SECRET,
+  FOOTER_TEXT,
+  LAST_COMMIT,
+  LAST_COMMIT_GH,
+  LAST_TAG,
+  LAST_TAG_GH,
+  LOG_LEVEL = 'info',
+  NODE_ENV = 'development',
+  OIDC_WELL_KNOWN_CONFIGURATION_URL,
+  PERSISTENT_BACKEND = 'preview',
+  PHASE = 'beta',
+  PORT = '3000',
+  PREVIEW_URL = 'http://dev.cdp-int.defra.cloud/forms-runner/',
+  PUBLISH_URL = 'http://dev.cdp-int.defra.cloud/forms-runner/',
+  REDIS_HOST = 'localhost',
+  REDIS_KEY_PREFIX = 'forms-designer',
+  REDIS_PASSWORD = 'my-password',
+  REDIS_USERNAME = 'default',
+  SESSION_COOKIE_PASSWORD = '',
+  SESSION_COOKIE_TTL,
+  SESSION_TTL,
+  S3_BUCKET
+} = process.env
+
 // Build config
-const config = {
-  port: process.env.PORT,
-  env: process.env.NODE_ENV,
-  appPathPrefix: process.env.APP_PATH_PREFIX || '/forms-designer',
-  previewUrl:
-    process.env.PREVIEW_URL || 'http://dev.cdp-int.defra.cloud/forms-runner/', // TODO set this to localhost and pull env vars from CDP
-  publishUrl:
-    process.env.PUBLISH_URL || 'http://dev.cdp-int.defra.cloud/forms-runner/', // TODO set this to localhost and pull env vars from CDP
-  persistentBackend: process.env.PERSISTENT_BACKEND || 'preview',
+const config: Config = {
+  port: parseInt(PORT),
+  env: NODE_ENV as Config['env'],
+  appDir: resolve(dirname(configPath), '../dist'),
+  appPathPrefix: APP_PATH_PREFIX,
+  clientDir: resolve(dirname(configPath), '../../client/dist'),
+  previewUrl: PREVIEW_URL,
+  publishUrl: PUBLISH_URL,
+  persistentBackend: PERSISTENT_BACKEND as Config['persistentBackend'],
   serviceName: 'Defra Form Builder',
-  s3Bucket: process.env.S3_BUCKET,
-  logLevel: process.env.LOG_LEVEL || 'info',
-  phase: process.env.PHASE || 'beta',
-  footerText: process.env.FOOTER_TEXT,
-  lastCommit: process.env.LAST_COMMIT || process.env.LAST_COMMIT_GH,
-  lastTag: process.env.LAST_TAG || process.env.LAST_TAG_GH,
-  sessionTimeout: process.env.SESSION_TIMEOUT,
-  sessionCookiePassword: process.env.SESSION_COOKIE_PASSWORD,
-  sessionCookieTtl: process.env.SESSION_COOKIE_TTL ?? '1800',
-  azureClientId: process.env.AZURE_CLIENT_ID,
-  azureClientSecret: process.env.AZURE_CLIENT_SECRET,
-  oidcWellKnownConfigurationUrl: process.env.OIDC_WELL_KNOWN_CONFIGURATION_URL,
-  appBaseUrl: process.env.APP_BASE_URL,
-  useSingleInstanceCache: process.env.USE_SINGLE_INSTANCE_CACHE,
-  redisHost: process.env.REDIS_HOST,
-  redisUsername: process.env.REDIS_USERNAME,
-  redisPassword: process.env.REDIS_PASSWORD,
-  redisKeyPrefix: process.env.REDIS_KEY_PREFIX,
-  redisTtl: process.env.REDIS_TTL
+  s3Bucket: S3_BUCKET,
+  logLevel: LOG_LEVEL as Config['logLevel'],
+  phase: PHASE as Config['phase'],
+  footerText: FOOTER_TEXT,
+  isProduction: NODE_ENV === 'production',
+  isDevelopment: !['production', 'test'].includes(NODE_ENV),
+  isTest: NODE_ENV === 'test',
+  lastCommit: LAST_COMMIT ?? LAST_COMMIT_GH,
+  lastTag: LAST_TAG ?? LAST_TAG_GH,
+  sessionTtl: SESSION_TTL
+    ? parseInt(SESSION_TTL)
```

Comment:
Same `parseInt` here.

Also, defaults/fallback values like this are better applied using `joi.default()` for consistency.

TBH I try and avoid config default values - they can become troublesome.